source/Audio.cpp

```diff
@@ -576,6 +576,10 @@ namespace {
 					return;
 				name = loadQueue.begin()->first;
 				path = loadQueue.begin()->second;
+
+				// Since we need to unlock the mutex below, create the map entry to
+				// avoid a race condition when accessing sounds' size.
+				sounds[name];
```

Comment:
It seems appropriate to also store the created/looked-up entry in a variable and use it on line 586 (line 582 in the original code) to avoid the need to accessing the map again.

Reply:
I guess. Didn't think it was necessary since I doubt that it will have an impact on performance (since it's already race free). thx :)